pkg/network/state_test.go

```diff
@@ -1145,8 +1147,14 @@ func TestDNSStatsWithMultipleClients(t *testing.T) {
 	}
 
 	dKey := dnsKey{clientIP: c.Source, clientPort: c.SPort, serverIP: c.Dest, protocol: c.Type}
-	stats := make(map[dnsKey]dnsStats)
-	stats[dKey] = dnsStats{successfulResponses: 1}
+
+	getStats := func() map[dnsKey]dnsStats {
+		stats := make(map[dnsKey]dnsStats)
+		countByRcode := make(map[uint8]uint32)
+		countByRcode[uint8(layers.DNSResponseCodeNoErr)] = 1
+		stats[dKey] = dnsStats{countByRcode: countByRcode}
+		return stats
+	}
```

Comment:
Why do we need to keep copying `stats` now?

Reply:
`stats` now has a `map` as its member. So if we reuse the same `stats` object, the map would get modified across different calls to `state.Connections`.